mkosi

```diff
@@ -380,7 +381,12 @@ def complete_step(text: str, text2: Optional[str] = None) -> Iterator[List[Any]]
     print_step(text2.format(*args) + '.')
 
 
-@complete_step('Detaching namespace')
+# https://github.com/python/mypy/issues/1317
+C = TypeVar('C', bound=Callable)
+completestep = cast(Callable[[str], Callable[[C], C]], complete_step)
```

Comment:
Alternative suggestion: rename the `def` to `completestep`, and make `complete_step` the cast version of it, so all the uses of the decorator wouldn’t need to be adjusted. I feel like that would be better, but I’m not sure, so for now that’s just a suggestion :)

Reply:
It's also used as a generator inside of `with` statements.  I'm not sure in which case it's used more, but I figured the decorator usage was easier to search/replace for.